src/libstd/sys/sgx/mod.rs

```diff
@@ -8,27 +8,24 @@ use os::raw::c_char;
 use sync::atomic::{AtomicBool, Ordering};
 
 pub mod abi;
-mod waitqueue;
```

Comment:
I'm slightly concerned by the large number of deletions in this module, can you explain how you've implemented all these primitives on this platform?

Reply:
`parking_lot` does not have that much platform specific code in it. It's basically just the thread parker code. And it uses the pthread based `unix` module for sgx it seems (it does for all unixes except Linux if you are running nightly)

Reply:
@faern sgx doesn't have pthread, and neither do other targets that don't use the code in sys/unix.

Reply:
Someone will need to port the `thread_parker` module in parking_lot to support SGX. At the moment any unsupported platforms will just fall back to a spinlock.

Reply:
I think reverting to an implementation that is not scheduler-aware is an unacceptable regression. Can the old synchronization primitives be retained for platforms not supported by `parking_lot`?

I'm also somewhat bothered that you are proposing to regress these platforms without so much as mentioning the relevant platform maintainers. I only found out about this PR by bascically randomly browsing Rust community websites. cc @EdSchouten @jackpot51 @ids1024

Reply:
I don't think anyone intend to introduce such regressions. So far we are playing around with using `parking_lot` and we have not yet fully figured out what that means. I agree that would be unacceptable.

I can look into porting the sgx code into a `parking_lot` thread parker implementation. I think that would be the best solution. It helps `parking_lot` itself improve and we don't need so much conditionally compiled code in libstd.

Reply:
I looked at the code, I think an implementation shouldn't be too difficult. You just need to call `wait(EV_UNPARK)` and `send(EV_UNPARK, id)` ([docs](https://edp.fortanix.com/docs/api/std/os/fortanix_sgx/usercalls/index.html)). I filed https://github.com/rust-lang/rust/pull/58949 which I think you need.

Reply:
Thanks for the input. I will look at it when I have time. But you are right, that I don't have an already exposed way to get the thread id. I'll follow the other PR to see how that comes along.

Reply:
Thanks for pointing this out to begin with. Currently working on adding `ThreadParker`s to the platforms that fall back to the spin lock. The SGX one is now merged, so this thread can be resolved I think.